### ISSUE TITLE
Feature request: `scss/function-no-unknown`

### ISSUE THREAD

**Issue by rdebeasi** [stylelint/stylelint#5784]
### What is the problem you're trying to solve?

Sass doesn't catch typos in function names. The docs [recommend](https://sass-lang.com/documentation/at-rules/function#plain-css-functions) using a linter to catch these issues:

>Because any unknown function will be compiled to CSS, it’s easy to miss when you typo a function name. Consider running a [CSS linter](https://stylelint.io/) on your stylesheet’s output to be notified when this happens!

I'm unable to find any rules in stylelint-css that catch mistakes like these, however.

### What solution would you like to see?

It would be awesome to have a new rule to catch mistakes like these. It would work like `scss/at-rule-no-unknown`, but for functions. Let's call it `scss/function-no-unknown`. 

Here's an example of a mistake. I've typed `invart` instead of `invert`:
```scss
@function invert($color, $amount: 100%) {
  $inverse: change-color($color, $hue: hue($color) + 180);
  @return mix($inverse, $color, $amount);
}

.button {
  background-color: invart(#f00)
}
```

Sass assumes that `invart` must be a CSS function and writes it into the compiled CSS:

```css
.button {
  background-color: invart(#f00)
}
```

This proposed rule would realize that there's no Sass function called `invart` and also no CSS function called `invart`. It would throw an error, alerting us to the fact that we've made a typo.

Thanks so much for your time!! 😄

**Comment by rdebeasi** [comment]
I inadvertently opened this issue in `stylelint` instead of `stylelint-scss`. Closing in favor of https://github.com/stylelint-scss/stylelint-scss/issues/581. Thanks!
